### PR TITLE
Use map format by default in 18.0 or over

### PR DESCRIPTION
**Pull request by mururu**

### CODE REVIEW

File: src/msgpack.erl [hunk 1]
```diff
@@ -165,6 +165,24 @@ parse_options([{ext, {Packer,Unpacker}}|TL], Opt0) when
 -ifdef(TEST).
 -include_lib("eunit/include/eunit.hrl").
 
+-ifdef(default_map).
```
Comment (kuenishi) [inline src/msgpack.erl]: What is this ifdef for? Looks like `test_data/0` is totally same.

File: test/msgpack_test.erl [hunk 2]
```diff
@@ -176,6 +176,25 @@ string_test() ->
...
+    [
+     {"pack",
+      fun() ->
+              Map = {[{1,2}]},
+              ?assertEqual(pack(Map, [{format, map}]), pack(Map))
+      end},
+     {"unpack",
+      fun() ->
+              Map = {[{1,2}]},
+              Binary = pack(Map, [{format, map}]),
```
Comment (kuenishi) [inline test/msgpack_test.erl]: Use `Binary = pack(Map, [{format, ?DEFAULT_INTERFACE}]),` instead. Maybe the name "default interface" is not comprehensive, alternate name like `?DEFAULT_MAP_STYLE` might be better. (Although the name `interface` in the record was not a bit appropriate)

File: test/msgpack_test.erl [hunk 3]
```diff
@@ -176,6 +176,25 @@ string_test() ->
     MsgpackStringBin = msgpack:pack(String),
     {ok, String} = msgpack:unpack(MsgpackStringBin).
 
+-ifdef(default_map).
```
Comment (kuenishi) [inline test/msgpack_test.erl]: And then you can remove this ifdef.

**Comment by kuenishi** [comment]
Thank you for contribution, although for now I am -1 for merge as it is as I commented above.

File: test/msgpack_test.erl (hunk 2 above)
Comment (mururu) [inline test/msgpack_test.erl]: How about `?DEFAULT_MAP_FORMAT` ?

File: src/msgpack.erl (hunk 1 above)
Comment (mururu) [inline src/msgpack.erl]: The difference is `nil`/`null`.

File: test/msgpack_test.erl (hunk 2 above)
Comment (kuenishi) [inline test/msgpack_test.erl]: That is cool.

File: src/msgpack.erl (hunk 1 above)
Comment (kuenishi) [inline src/msgpack.erl]: Okay, fine. It looks like it's a good change to rethink on `nil/null` stuff, which is very confusing.

**Comment by kuenishi** [comment]
After the default map format macro updated, I'll merge this. Thank you.

**Comment by mururu** [comment]
Thanks. Updated.

**Pull request merged**